### ISSUE TITLE
Add welcome screen

### ISSUE THREAD

**Issue by Bodigrim** [haskell/play-haskell#52]
### Feature request

Currently if an uninitiated user hits https://play.haskell.org there is no introduction / welcome screen, just a sheer wall of code. This could be frightening for newcomers. Could it be possible to detect if it's a first visit to the domain and display a modal window, describing purpose / capabilities of the site and where to go next.

**Comment by tomsmeding** [comment]
That's possible. Neither the [Rust playground](https://play.rust-lang.org/) nor the [Go playground](https://play.golang.com/) seem to have this -- at least, not when I visit them in incognito mode.

There is already a `Help` button in the top-left of the screen; infelicities of that thing are 1. the placement of the modal is not ideal on mobile or small screens, either `<dialog>` is uncooperative or my CSS skills are not strong enough; 2. there is no introductory sentence at the top of the modal; 3. the Help button is not immediately visible on mobile, you have to scroll the top toolbar to bring it in view.

Could we repurpose the Help screen for this? What text / design did you have in mind?

**Comment by Bodigrim** [comment]
Yes, if we can make Help screen to pop up at first visit, it would be great. 

A few introductory sentences; what this playground is capable of / suitable for, and what it's not sufficient for; where to go next if you need more (basically a link to `ghcup`).

**Comment by Bodigrim** [comment]
If you are past Playground capabilities you are already capable of "Running your first lines of code" and "Writing your first Haskell program", so I'd send users straight up to https://www.haskell.org/ghcup/#.

(I think it's rather https://www.haskell.org/get-started/ should send people to Playground first instead of suggesting to install local development environment)

**Comment by hasufell** [comment]
I'm not sure I am a fan of popups when I visit websites.

I think a help button should be enough.

**Comment by tomsmeding** [comment]
> I'm not sure I am a fan of popups when I visit websites.

Agreed.

> I think a help button should be enough.

What do you think of the placement of the button currently? It's reachable and visible on desktop, but takes scrolling the toolbar on mobile. (And also I need to fix the popup on mobile anyway...)

**Comment by hasufell** [comment]
> > I'm not sure I am a fan of popups when I visit websites.
> 
> Agreed.
> 
> > I think a help button should be enough.
> 
> What do you think of the placement of the button currently? It's reachable and visible on desktop, but takes scrolling the toolbar on mobile or small screens. (And also I need to fix the popup on mobile anyway...)

![Screenshot_20241005_175934.jpg](https://github.com/user-attachments/assets/a40bae78-d5b2-4c2a-b131-5317198c3d0c)

Maybe a question mark symbol there?

Alternatively a small question mark symbol directly on the haskell icon. Like notification style. It would vanish if you're not a new time visitor and the original help text button would persist.

**Comment by tomsmeding** [comment]
Where is the "there" you mean?

**Comment by hasufell** [comment]
> Where is the "there" you mean?

I reuploaded the picture
